composables/useIdentity.ts

```diff
@@ -26,74 +30,72 @@ export default function useIdentity({
   customNameOption?: string
 }) {
   const { urlPrefix } = usePrefix()
-  const { apiInstanceByPrefix } = useApi()
-  const { isDotAddress, getIdentityId, clientName } =
-    useIdentityQuery(urlPrefix)
+  const { getIdentityId } = useIdentityQuery(urlPrefix)
 
   const id = computed(() => address.value && getIdentityId(address.value))
 
-  const identityPrefix = computed(() => (isDotAddress.value ? 'dot' : 'rmrk'))
-
-  const identityUnit = computed(
-    () => chainPropListOf(identityPrefix.value)?.tokenSymbol,
-  )
-
-  const identityApi = computed(() => apiInstanceByPrefix(identityPrefix.value))
-
-  const { data, refetch, loading } = useGraphql({
-    clientName,
-    queryName: 'identityById',
-    variables: {
-      id: id.value,
-    },
-    disabled: computed(() => !address.value),
-  })
-
-  const identity = computed<IdentityFields>(() => data.value?.identity || {})
+  const { profile, pending: loading } = useFetchProfile(id)
 
   const shortenedAddress = computed(() => shortAddress(address.value))
-  const twitter = computed(() => identity?.value?.twitter)
-  const display = computed(() => identity?.value?.display)
+  const twitter = computed(
+    () =>
+      profile?.value?.socials?.find((s) => s.platform === 'Twitter')?.handle,
+  )
+  const web = computed(
+    () =>
+      profile?.value?.socials?.find((s) => s.platform === 'Website')?.handle,
+  )
+  const farcaster = computed(
+    () =>
+      profile?.value?.socials?.find((s) => s.platform === 'Farcaster')?.handle,
+  )
```

Comment:
maybe like this nicer?

```suggestion
const getSocialHandle = (platform) => computed(() =>
  profile?.value?.socials?.find((s) => s.platform === platform)?.handle,
);

const twitter = getSocialHandle('Twitter');
const web = getSocialHandle('Website');
const farcaster = getSocialHandle('Farcaster');
```